internal/resource/config/oauth/oauth_auth_server_settings_data_source.go

```diff
@@ -0,0 +1,620 @@
+	state.PersistentGrantIdleTimeout = types.Int64PointerValue(r.PersistentGrantIdleTimeout)
+	state.PersistentGrantIdleTimeoutTimeUnit = types.StringPointerValue(r.PersistentGrantIdleTimeoutTimeUnit)
+	state.RefreshTokenLength = types.Int64Value(r.RefreshTokenLength)
+	state.RollRefreshTokenValues = types.BoolPointerValue(r.RollRefreshTokenValues)
+	state.RefreshTokenRollingGracePeriod = types.Int64PointerValue(r.RefreshTokenRollingGracePeriod)
+	state.RefreshRollingInterval = types.Int64Value(r.RefreshRollingInterval)
+	state.PersistentGrantReuseGrantTypes = internaltypes.GetStringSet(r.PersistentGrantReuseGrantTypes)
+	state.BypassAuthorizationForApprovedGrants = types.BoolPointerValue(r.BypassAuthorizationForApprovedGrants)
+	state.AllowUnidentifiedClientROCreds = types.BoolPointerValue(r.AllowUnidentifiedClientROCreds)
+	state.AllowUnidentifiedClientExtensionGrants = types.BoolPointerValue(r.AllowUnidentifiedClientExtensionGrants)
+	state.AdminWebServicePcvRef, respDiags = resourcelink.ToDataSourceState(ctx, r.AdminWebServicePcvRef)
+	diags.Append(respDiags...)
+	state.AtmIdForOAuthGrantManagement = types.StringPointerValue(r.AtmIdForOAuthGrantManagement)
+	state.ScopeForOAuthGrantManagement = types.StringPointerValue(r.ScopeForOAuthGrantManagement)
+	state.AllowedOrigins = internaltypes.GetStringSet(r.AllowedOrigins)
+	state.UserAuthorizationUrl = types.StringPointerValue(r.UserAuthorizationUrl)
+	state.RegisteredAuthorizationPath = types.StringValue(r.RegisteredAuthorizationPath)
+	state.PendingAuthorizationTimeout = types.Int64Value(r.PendingAuthorizationTimeout)
+	state.DevicePollingInterval = types.Int64Value(r.DevicePollingInterval)
+	state.ActivationCodeCheckMode = types.StringPointerValue(r.ActivationCodeCheckMode)
+	state.BypassActivationCodeConfirmation = types.BoolValue(r.BypassActivationCodeConfirmation)
+	state.UserAuthorizationConsentPageSetting = types.StringPointerValue(r.UserAuthorizationConsentPageSetting)
+	state.UserAuthorizationConsentAdapter = types.StringPointerValue(r.UserAuthorizationConsentAdapter)
+	state.ApprovedScopesAttribute = types.StringPointerValue(r.ApprovedScopesAttribute)
+	state.ApprovedAuthorizationDetailAttribute = types.StringPointerValue(r.ApprovedAuthorizationDetailAttribute)
+	state.ParReferenceTimeout = types.Int64PointerValue(r.ParReferenceTimeout)
+	state.ParReferenceLength = types.Int64PointerValue(r.ParReferenceLength)
+	state.ParStatus = types.StringPointerValue(r.ParStatus)
+	state.ClientSecretRetentionPeriod = types.Int64PointerValue(r.ClientSecretRetentionPeriod)
+	state.JwtSecuredAuthorizationResponseModeLifetime = types.Int64PointerValue(r.JwtSecuredAuthorizationResponseModeLifetime)
+	return diags
+}
+
+// Read resource information
+func (r *oauthAuthServerSettingsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state oauthAuthServerSettingsDataSourceModel
+
+	diags := req.Config.Get(ctx, &state)
```

Comment:
Didn't notice this before - there's no need to run this req.Config.Get because we don't need anything from the Config to retrieve this resource.